src/cmdline/mod.rs

```diff
@@ -336,172 +250,669 @@ impl Cmdline {
     }
 }
 
+impl UpdateArgs for BootArgs {
+    fn has_capacity(&self, more: usize) -> bool {
+        let mut auxiliary_space = 0;
+
+        if !self.0.is_empty() {
+            // this is NOT the first arg so we need to add an extra space between existing
+            // args and the new args to be added
+            auxiliary_space += 1;
+        }
+
+        self.0.len() + auxiliary_space + more <= CMDLINE_MAXSIZE
+    }
+
+    fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()> {
+        let s = slug.as_ref().trim();
+        valid_str(s)?;
+
+        if !self.has_capacity(s.len()) {
+            return Err(Error::TooLarge);
+        }
+
+        if !self.0.is_empty() {
+            self.0.push(' ');
+        }
+
+        self.0.push_str(s);
+
+        Ok(())
+    }
+}
+
+/// An abstraction for the init arguments of a kernel command line.
+#[derive(Clone, Debug)]
+pub struct InitArgs(String);
+
+impl UpdateArgs for InitArgs {
+    fn has_capacity(&self, more: usize) -> bool {
+        let mut auxiliary_space = 0;
+
+        if !self.0.is_empty() {
+            // this is NOT the first arg so we need to add an extra space between existing
+            // args and the new args to be added
+            auxiliary_space += 1;
```

Comment:
Shouldn't it be enough to initialize this to 1 here instead of adding 1?

Reply:
simplifying even further and removing mutability by refactoring into: let auxiliary_space = if self.is_empty() { 0 } else { 1 };